### ISSUE TITLE
Image with multiple SDKs

### ISSUE THREAD

**Issue by Kralizek** [microsoft/dotnet-framework-docker#731]
### Describe the Problem

More and more CI tools like CircleCI and AWS CodeBuild are based on executing the process in containers. This works really fine when creating applications that are bound to a single target framework.

The same unfortunately can't be said when packaging a library that targets multiple frameworks, especially when executing the tests.

An example is a solution with
- library targeting net48, netstandard2.1, net5.0
- test project targeting net48, netcoreapp3.1, net5.0

In cases like this, you need to orchestrate a build across multiple containers. While working, you lose the advantages of having artifacts and test results coming from the same execution (this is a problem i am experiencing with CircleCI.

### Describe the Solution

Ideally it would be nice if there were images that included multiple runtimes and SDKs.

Best options would be
- an image including selected version of netfx and all netcore lts and latest current (currently 3.1 and 5.0). This image would obviously work only on windows.
- an image including only netcore runtimes that can be used on linux

Alternatively, it would be acceptable for me if you could publish a guide with a series of concrete steps to achieve the same.

### Additional Context

I was encouraged to open this issue from a conversation in Twitter: https://twitter.com/Kralizek/status/1360295293426806793

**Comment by MichaelSimons** [comment]
[Triage] This topic of multi-purpose images is an interesting to us.

We could add the multiple runtimes to support this scenario.  The downside is that it adds more to an already large image.  We don't have a sense for how many users would make use of the multiple runtimes.

We think a multi-stage Dockerfile could be utilized here.  The idea is to use different stages specific to each target framework.  We would like to see someone create a proof of concept that illustrates this idea and possibly incorporate it into our samples.

**Comment by lukos** [comment]
The most obvious way is to build out your own images with all of the targets you want installed. After reading another related issue and from my own experience, I think the problem is the number of different ways to do this and it not being clear the correct way.

For example, you can install things with chocolatey, nuget, powershell, msi etc. and the urls to obtained some of the msi's change over time.

If we could find a really consistent way to do this then the clean way would be something like:

```dockerfile
FROM mcr.microsoft.com/dotnet/framework/sdk:4.7.2 AS base

SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop';$ProgressPreference='silentlyContinue';"]
Add-AppxPackage https://consistently.named.url/netsdk/sdk48.msix
Add-AppxPackage https://consistently.named.url/netsdk/sdknetstandard21.msix
Add-AppxPackage https://consistently.named.url/netsdk/sdk5.msix
```

Note that I am also opposed to just inventing _another_ new thing that just makes all of this even more confusing!

**Comment by slonopotamus** [comment]
So, what's wrong with `choco install netfx-4.7.2-devpack + <whatever else packages you need>`?

**Comment by Kralizek** [comment]
I guess getting where "choco install" actually works and doesn't throw an error? 🤔

**Comment by slonopotamus** [comment]
This works for me without any errors:
```dockerfile
FROM mcr.microsoft.com/windows/servercore:20H2
RUN powershell -NoProfile -ExecutionPolicy Bypass -Command "Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))"
RUN choco install -y netfx-4.7.2-devpack
```
What error are you talking about?

**Comment by slonopotamus** [comment]
There's an alternative way: use VS Build Tools (or even Visual Studio) installer, since you most likely would need things like MSBuild and others from it anyway:

```dockerfile
FROM mcr.microsoft.com/windows/servercore:20H2
ADD https://aka.ms/vs/16/release/vs_buildtools.exe C:/
RUN C:/vs_buildtools.exe --quiet --wait --nocache \
	--add Microsoft.Net.Component.4.7.2.SDK \
	--add Microsoft.Net.Component.4.7.2.TargetingPack
```

Refer to [docs](https://docs.microsoft.com/en-us/visualstudio/install/workload-component-id-vs-build-tools?view=vs-2019) for a list of available component ids.

**Comment by lbussell** [comment]
@mthalman was this resolved by https://github.com/microsoft/dotnet-framework-docker/pull/845?

.NET Framework SDK images include `Microsoft.NetCore.Component.SDK` and all supported `Microsoft.NetCore.Component.Runtime.*` components.

**Comment by mthalman** [comment]
Yes, I would say this is resolved. It includes the latest SDK (.NET 9) which provides support for down-level TFMs (e.g. `net8.0`). So installing multiple SDK versions isn't necessary. And it does include multiple runtime versions.
